src/types.ts

```diff
@@ -395,6 +394,7 @@ export interface RangeTime {
 export interface UnavailableHour {
   start: number;
   end: number;
+  [key: string]: any;
```

Comment:
@Fvaco Can we type this a little bit better maybe? 😄

Reply:
Not sure 🤔 , this is the type that accepts literally anything that we'll send to the `CustomUnavailabilityItem` component, I don't know how to narrow down the typing without limiting the flexibility of thiese props. Any idea?

Reply:
We can add **T** as a generic type for **UnavailableHour**.  

```
export interface UnavailableHour <T> {
  start: number;
  end: number;
  [key: string]: T;
}
```

Which will not be super necessary for our specific project as we are using javascript but maybe if we are going to reuse in the future with other typescript projects will be better. 😄 
We don't have strong opinion on the usage of any so I'm fine anyway if we decide to keep it

Reply:
That generic isn't super useful, right? With that T, we could do something like:

`UnavailableHour<number>`

Which would force us to use `number` for each and every custom prop passed. Our goal is being able to add any kind of extra prop to the `CustomUnavailableItem`

What would work, though is doing something like

```jsx
export interface UnavailableHour <T = { [key:string]: any }> { 
    start: number; 
    end: number; 
    extraData: T; 
}
```
Which would many other changes but would let us set any type for `extraData`.


Please correct me if I'm wrong. Am I missing something important about generics? 🤔

Reply:
I think we can use it as T without giving information about the **[key: string]: any** because in the end we can pass it in the format that we want. 

```
export interface UnavailableHour <T> { 
    start: number; 
    end: number; 
    extraData: T; 
}

```
I think maybe adding the **extraData** can more clear than adding the custom props, for me it's just a matter of not using **any** and give a little bit more of typing 😄

Reply:
Yep, you're right! I'll give it a go using the extraData key! If the number of needed changes is too high we could come up with an alternative Idea like joining generics or something a bit smarter. The `= { [key:string]: any }` part would avoid forcing every part of the code to use the generic even if it doesn't include extra data. Just a retro-compatibility thing 😄 

Without the default value
![Captura de pantalla 2023-11-20 a las 13 12 59](https://github.com/DocPlanner/react-native-calendar-kit/assets/4592491/3d985bc4-2fdb-48cb-96f4-c85dbfd71066)

With the default value:
![Captura de pantalla 2023-11-20 a las 13 15 15](https://github.com/DocPlanner/react-native-calendar-kit/assets/4592491/84a9f821-585f-4533-b612-1c803a72844d)